source/Mlos.Python/mlos/Optimizers/OptimizationProblem.py

```diff
@@ -99,19 +104,12 @@ def __init__(
 
         # Fit functions / surrogate models will be fed features consisting of both context and parameters.
         # Thus, the feature space is comprised of both context and parameters.
-        self.feature_space = SimpleHypergrid(
-            name="features",
-            dimensions=[
-                CategoricalDimension(name="contains_parameters", values=[True]),
-                CategoricalDimension(name="contains_context", values=[self.context_space is not None])
-            ]
-        ).join(
-            subgrid=self.parameter_space,
-            on_external_dimension=CategoricalDimension(name="contains_parameters", values=[True])
-        ).join(
-            subgrid=self.context_space,
-            on_external_dimension=CategoricalDimension(name="contains_context", values=[True])
-        )
+        parameter_names = {parameter.name for parameter in parameter_space.root_dimensions}
+        context_names = [context.name for context in context_space.root_dimensions]
+        if parameter_names.intersection(context_names):
+            raise ValueError("Parameter space and context space can't share dimensions.")
+        self.feature_space = SimpleHypergrid(name='features',
+                                             dimensions=[*parameter_space.dimensions, *context_space.dimensions])

```

Comment:
I think it would be better to perform a Hypergrid.join() here so that each feature is prefixed with either `parameter.` or `context.` It will prevent name collisions, as well as ease debugging and data analysis since these prefixes will be propagated all the way to the csv files and to JMP for @edcthayer to see. So he won't need to wonder which columns are parameters and which are context.

Also, it's plausible that context features will contain benchmark parameters, and these could easily collide with target parameters.

Reply:
I vaguely remember that it was a bit awkward to do that but let me try again. Not sure any more what issue I ran into.